util/main.cpp

```diff
@@ -0,0 +1,251 @@
+    "  -e, --ecc          Cut out ECC data from source file\n"
+    "\n"
+    "Output options:\n"
+    "  -n, --number=NUM   Print only section with number NUM, this option  can be\n"
+    "                     specified multiple times, not applicable for JSON output\n"
+    "  -o, --output=FMT   Set output format:\n"
+    "                       table  formatted output, used by default\n"
+    "                       long   long lines without splitting\n"
+    "                       json   JSON output\n"
+    "                       hex    hex dump of payload\n"
+    "                       bin    binary data of payload\n"
+    "Parser setup:\n"
+    "  --fsp-trace=FILE   Set path to FSP trace utility [" DEFAULT_FSP_TRACE "]\n"
+    "  --occ-str=FILE     Set path to OCC string file [" DEFAULT_OCC_STRINGS "]\n"
+    "  --hb-str=FILE      Set path to HostBoot string file [" DEFAULT_HB_STRINGS "]\n"
+    "  --hb-sym=FILE      Set path to HostBoot symbols file [" DEFAULT_HB_SYMBOLS "]\n"
+    "\n"
+    "Other options:\n"
+    "  -v, --version      Print version and exit\n"
+    "  -h, --help         Print this help and exit\n"
+    );
+    // clang-format on
+}
+
+/** @brief Application entry point. */
+int main(int argc, char* argv[])
+{
+    int optFlag = 0;
+    // clang-format off
+    const struct option opts[] = {
+        { "file",       required_argument, nullptr,  'f' },
+        { "bmc",        required_argument, nullptr,  'b' },
+        { "bmc-list",   no_argument,       &optFlag, OptBmcList },
+        { "pnor",       required_argument, nullptr,  'p' },
+        { "pnor-list",  no_argument,       &optFlag, OptPnorList },
+        { "hbel",       required_argument, &optFlag, OptHbelDump },
+        { "ecc",        no_argument,       nullptr,  'e' },
+        { "number",     required_argument, nullptr,  'n' },
+        { "output",     required_argument, nullptr,  'o' },
+        { "fsp-trace",  required_argument, &optFlag, OptFspTrace },
+        { "occ-str",    required_argument, &optFlag, OptOCCStr },
+        { "hb-str",     required_argument, &optFlag, OptHbStr },
+        { "hb-sym",     required_argument, &optFlag, OptHbSym },
+        { "version",    no_argument,       nullptr,  'v' },
+        { "help",       no_argument,       nullptr,  'h' },
+        { 0, 0, nullptr, 0 }
+    };
+    // clang-format on
+
+    // Short options
+    char shortOpts[(sizeof(opts) / sizeof(opts[0])) * 2 + 1];
+    char* shortOptPtr = shortOpts;
+    for (size_t i = 0; i < sizeof(opts) / sizeof(opts[0]); ++i)
+    {
+        if (!opts[i].flag)
+        {
+            *shortOptPtr++ = opts[i].val;
+            if (opts[i].has_arg != no_argument)
+                *shortOptPtr++ = ':';
+        }
+    }
+
+    Printer printer;
+    Task task(printer);
+
+    // Parse arguments
+    opterr = 0;
+    int optVal;
+    while ((optVal = getopt_long(argc, argv, shortOpts, opts, nullptr)) != -1)
```

Comment:
CLIUitls/CLI11 is very fashionable in OpenBMC :)

Reply:
imho, it is not a good idea to use cli11. I prefer function call to macro.